packages/flutter/lib/src/material/flat_button.dart

```diff
@@ -109,11 +111,14 @@ class FlatButton extends MaterialButton {
     Brightness colorBrightness,
     EdgeInsetsGeometry padding,
     ShapeBorder shape,
-    Clip clipBehavior,
+    Clip clipBehavior = Clip.none,
```

Comment:
Here and elsewhere.

This is a slightly incompatible change. The old docs that said "clipBehavior must not be null" were wrong; could have been removed. It's incompatible because apps could safely pass along `clipBehavior: null`, despite the docs. It's unlikely to affect many apps of course.

Reply:
I'm going to revert this in this PR and submit a different PR that normalizes this as we discussed offline (all all uses to accept null, and default to `Clip.none` at the lowest level using `??`)